File: samples/csharp/src/Queues.cs
using System;
using System.Collections.Generic;

// UI settings
// Expression: q
//   Type: System.Collections.Generic.Queue<string>
//     Layout: Queue
//     Save: Object Attributes

internal class Queues
{
    public static void RunMain(string[] args)
    {
        new Queues().Start();
    }

    public void Start()
    {
        Queue<string> q = new Queue<string>();
        q.Enqueue("1");
        q.Enqueue("3");
        q.Enqueue("2");
        q.Enqueue("4");

        while (q.Count > 0)
        {
            string node = q.Dequeue();
            Console.WriteLine(node);
        }

        Queue<QueueNode<string, int>> qo = new Queue<QueueNode<string, int>>();
        qo.Enqueue(new QueueNode<string, int>("1", 1));
        qo.Enqueue(new QueueNode<string, int>("3", 3));
        qo.Enqueue(new QueueNode<string, int>("2", 2));
        qo.Enqueue(new QueueNode<string, int>("4", 4));

        while (qo.Count > 0)
        {
            QueueNode<string, int> node = qo.Dequeue();
            Console.WriteLine(node.value + ":" + node.data);
        }

        PriorityQueue<string, int> pq = new PriorityQueue<string, int>();
        pq.Enqueue("1", 1);
        pq.Enqueue("3", 3);
        pq.Enqueue("2", 2);
        pq.Enqueue("4", 4);
        while (pq.Count > 0)
        {
            string node = pq.Dequeue();
            Console.WriteLine(node);
        }
        
        Console.WriteLine("done");
    }
}

class QueueNode<T, S>
{
    public T value;
    public S data;

    public QueueNode(T value, S data)
    {
        this.value = value;
        this.data = data;
    }

    public override string ToString()
    {
        return this.data + ":" + this.value;
    }
}

File: samples/python/src/Queues.py
from __future__ import annotations
from typing import TypeVar, Generic
from queue import PriorityQueue, Queue

T = TypeVar("T")
S = TypeVar("S")
P = TypeVar("P")

# UI settings
# Expression: q
#   Type: Queue
#     Layout: Queue
#     Save: Object Attributes

class Queues:

    @staticmethod
    def run_main():
        Queues().start()

    def start(self):
        q: Queue[str] = Queue[str]()
        q.put("1")
        q.put("3")
        q.put("2")
        q.put("4")

        while len(q.queue) > 0:
            node: str = q.get()
            print(node)

        qo: Queue[QueueNode[str, int]] = Queue[QueueNode[str, int]]()
        qo.put(QueueNode[str, int]("1", 1))
        qo.put(QueueNode[str, int]("3", 3))
        qo.put(QueueNode[str, int]("2", 2))
        qo.put(QueueNode[str, int]("4", 4))

        while len(qo.queue) > 0:
            node: QueueNode[str, int] = qo.get()
            print(str(node.value) + ":" + str(node.data))

        pq: PriorityQueue = PriorityQueue()
        pq.put(1, "1")
        pq.put(3, "3")
        pq.put(2, "2")
        pq.put(4, "4")

        while len(pq.queue) > 0:
            value = pq.get()
            print(value)

        print("done")

class QueueNode(Generic[T, S]):
    def __init__(self, value: T, data: S):
        self.value: T = value
        self.data: S = data

    def __hash__(self):
        return super().__hash__()

    def __str__(self):
        return str(self.data) + ":" + str(self.value)


Queues.run_main()
